### ISSUE TITLE
Upgrade from C129 -> C130 on macOS via 1.72.x - Nightly (Checking C130)

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#41324]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by LaurenWags** [comment]
Upgrade checks (using x64 .dmg):

```

```

and

```

```

Note, need to be on internal VPN to do the below upgrade checks.

<details>
<summary>C130 --> C130 on-demand upgrade check - InProgress</summary>

1. Install `C129` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1727798559900959)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
6. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates-panel.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
7. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

8. Confirm you are prompted to "Relaunch" to get update (note, if you click "Relaunch" button you will relaunch and could potentially pick up a newer Nightly that is on the previous Chromium version. Closing and then relaunching via the terminal will not update you again)

SS

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------

Confirmed deltas:

**`Example`** | **`Example`**
----------------|----------------

</details>

**Comment by GeetaSarvadnya** [comment]
Upgrade checks (using arm64 .dmg):

```

```

Note, need to be on internal VPN to do the below upgrade checks.

<details>
<summary>C130 --> C130 on-demand upgrade check - </summary>

1. Install `C129` PR build (arm64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1725919055620299?thread_ts=1725908115.169339&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
6. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates-panel.brave.com/sparkle/Brave-Browser/nightly-arm64/appcast.xml`
7. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```
[5330:259:0910/101508.505800:INFO:sparkle_glue.mm(327)] brave update: did finish loading appcast
[5330:259:0910/101508.507620:INFO:sparkle_glue.mm(333)] brave update: did find valid update with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[5330:259:0910/101508.509662:INFO:sparkle_glue.mm(354)] brave update: willDownloadUpdate with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[5330:259:0910/101517.411388:INFO:sparkle_glue.mm(390)] brave update: will install update on quit with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
```

8. Confirm you are prompted to "Relaunch" to get update (note, if you click "Relaunch" button you will relaunch and could potentially pick up a newer Nightly that is on the previous Chromium version. Closing and then relaunching via the terminal will not update you again)

![Screenshot 2024-09-10 at 10 15 44 AM](https://github.com/user-attachments/assets/f3f3161f-af7a-4f53-ab8d-04095a85c675)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2024-09-10 at 10 16 52 AM](https://github.com/user-attachments/assets/33def90e-b5a2-430f-b4e8-7e718ac652eb) | ![Screenshot 2024-09-10 at 10 17 00 AM](https://github.com/user-attachments/assets/22dab934-fbc0-4787-af95-5a8cfdd4a350)

Confirmed deltas:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2024-09-10 at 10 16 01 AM](https://github.com/user-attachments/assets/b8bdd986-ae59-4bb3-9023-c6dc3ae4e38e) | ![Screenshot 2024-09-10 at 10 16 14 AM](https://github.com/user-attachments/assets/ab7785ef-07bf-448f-a19c-f04e31340a14)

</details>

<details>
<summary>C129 --> C130 on-demand upgrade check - </summary>

1. Install `C128` build via https://github.com/brave/brave-browser/releases/tag/v1.71.72
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates-panel.brave.com/sparkle/Brave-Browser/nightly-arm64/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```
[5899:259:0910/102831.622039:INFO:sparkle_glue.mm(327)] brave update: did finish loading appcast
[5899:259:0910/102831.623783:INFO:sparkle_glue.mm(333)] brave update: did find valid update with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[5899:259:0910/102831.626641:INFO:sparkle_glue.mm(354)] brave update: willDownloadUpdate with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[5899:259:0910/102931.514162:INFO:sparkle_glue.mm(390)] brave update: will install update on quit with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
```

7. Confirm you are prompted to "Relaunch" to get update (see note in first case about this button)

![Screenshot 2024-09-10 at 10 29 59 AM](https://github.com/user-attachments/assets/2aaacc16-5663-4332-a4cf-9867ff16295e)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2024-09-10 at 10 30 51 AM](https://github.com/user-attachments/assets/7c645f5e-ae94-4ca6-9267-a218f57904c7) | ![Screenshot 2024-09-10 at 10 30 58 AM](https://github.com/user-attachments/assets/9840a76f-11b4-4210-98bc-d430c7360b85)

</details>

<details>
<summary>C130 --> C130 background upgrade check - </summary>

1. Install `C129` PR build (arm64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1725919055620299?thread_ts=1725908115.169339&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
8. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates-panel.brave.com/sparkle/Brave-Browser/nightly-arm64/appcast.xml`
9. DO NOT navigate to `brave://settings/help`
10. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```
[6700:259:0910/134731.767984:INFO:sparkle_glue.mm(327)] brave update: did finish loading appcast
[6700:259:0910/134731.769134:INFO:sparkle_glue.mm(333)] brave update: did find valid update with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[6700:259:0910/134731.770584:INFO:sparkle_glue.mm(354)] brave update: willDownloadUpdate with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[6700:259:0910/134739.366624:INFO:sparkle_glue.mm(390)] brave update: will install update on quit with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
```

11. Close and restart the browser via terminal with same command as step 8
12. Check version is updated under `brave://version`

![Screenshot 2024-09-10 at 1 49 06 PM](https://github.com/user-attachments/assets/fd6541a4-fcb3-4ec4-83f3-e8761f41e7f4)

13. Confirmed deltas:

Example | Example
--- | ---
![Screenshot 2024-09-10 at 1 48 31 PM](https://github.com/user-attachments/assets/0f319b07-2d76-447f-8eaa-a2f70a9441dc) | ![Screenshot 2024-09-10 at 1 48 42 PM](https://github.com/user-attachments/assets/6a2ef133-2745-435b-b392-ac1ebe6295fd)

</details>

<details>
<summary>C129 --> C130 background upgrade check - </summary>

1. Install `C128` build via https://github.com/brave/brave-browser/releases/tag/v1.71.72
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates-panel.brave.com/sparkle/Brave-Browser/nightly-arm64/appcast.xml`
9. DO NOT navigate to `brave://settings/help`
10. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```
[7133:259:0910/171232.955879:INFO:sparkle_glue.mm(327)] brave update: did finish loading appcast
[7133:259:0910/171232.958005:INFO:sparkle_glue.mm(333)] brave update: did find valid update with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[7133:259:0910/171232.959872:INFO:sparkle_glue.mm(354)] brave update: willDownloadUpdate with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
[7133:259:0910/171258.186405:INFO:sparkle_glue.mm(390)] brave update: will install update on quit with AppcastItem(Date: Tue, 10 Sep 2024 02:16:53 +0000, Version: 1.71.75.0)
```

11. Close and restart the browser via terminal with same command as step 7
12. Check version is updated under `brave://version`

![Screenshot 2024-09-10 at 5 13 50 PM](https://github.com/user-attachments/assets/e7006637-3200-4718-b762-e015800c1a77)

</details>
